sql/sqlstats/insights: TestIngester failed

sql/sqlstats/insights.TestIngester [failed](https://tanzanite.cluster.engflow.com/invocations/default/69cba537-0b33-4506-81ee-34815a3fb946?testReportRun=13&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NxbHN0YXRzL2luc2lnaHRzOmluc2lnaHRzX3Rlc3Q=) on release-25.1 @ [6123ebaa10c3847899aa81b09ec4cb8d08c0bd6d](https://github.com/cockroachdb/cockroach/commits/6123ebaa10c3847899aa81b09ec4cb8d08c0bd6d):


```
=== RUN   TestIngester
--- FAIL: TestIngester (1.67s)
=== RUN   TestIngester/One_Session
    ingester_test.go:90: 
        	Error Trace:	pkg/sql/sqlstats/insights/ingester_test.go:90
        	Error:      	Condition never satisfied
        	Test:       	TestIngester/One_Session
    --- FAIL: TestIngester/One_Session (1.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=13</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIngester.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46650

--- a/pkg/sql/sqlstats/insights/ingester.go
+++ b/pkg/sql/sqlstats/insights/ingester.go
@@ -16,6 +16,10 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
 )
 
+// defaultFlushInterval specifies a default for the amount of time an ingester
+// will go before flushing its contents to the registry.
+const defaultFlushInterval = time.Millisecond * 500
+
 // ConcurrentBufferIngester amortizes the locking cost of writing to an
 // insights registry concurrently from multiple goroutines. To that end, it
 // contains nothing specific to the insights domain; it is merely a bit of
@@ -29,6 +33,9 @@ type ConcurrentBufferIngester struct {
 	opts struct {
 		// noTimedFlush prevents time-triggered flushes from being scheduled.
 		noTimedFlush bool
+		// flushInterval is an optional override flush interval
+		// a value of zero will be set to the 500ms default.
+		flushInterval time.Duration
 	}
 
 	eventBufferCh chan eventBufChPayload
@@ -77,6 +84,13 @@ func WithoutTimedFlush() BufferOpt {
 	}
 }
 
+// WithFlushInterval allows for the override of the default flush interval
+func WithFlushInterval(intervalMS int) BufferOpt {
+	return func(i *ConcurrentBufferIngester) {
+		i.opts.flushInterval = time.Millisecond * time.Duration(intervalMS)
+	}
+}
+
 func (i *ConcurrentBufferIngester) Start(
 	ctx context.Context, stopper *stop.Stopper, opts ...BufferOpt,
 ) {
@@ -103,10 +117,14 @@ func (i *ConcurrentBufferIngester) Start(
 	})
 
 	if !i.opts.noTimedFlush {
+		flushInterval := i.opts.flushInterval
+		if flushInterval == 0 {
+			flushInterval = defaultFlushInterval
+		}
 		// This task eagerly flushes partial buffers into the channel, to avoid
 		// delays identifying insights in low-traffic clusters and tests.
 		_ = stopper.RunAsyncTask(ctx, "insights-ingester-flush", func(ctx context.Context) {
-			ticker := time.NewTicker(500 * time.Millisecond)
+			ticker := time.NewTicker(flushInterval)
 
 			for {
 				select {

--- a/pkg/sql/sqlstats/insights/ingester_test.go
+++ b/pkg/sql/sqlstats/insights/ingester_test.go
@@ -77,7 +77,7 @@ func TestIngester(t *testing.T) {
 				}, store, nil),
 			)
 
-			ingester.Start(ctx, stopper)
+			ingester.Start(ctx, stopper, WithFlushInterval(10))
 			for _, e := range tc.observations {
 				if e.statementID != 0 {
 					ingester.ObserveStatement(e.SessionID(), &Statement{ID: e.StatementID()})
@@ -93,7 +93,7 @@ func TestIngester(t *testing.T) {
 					numInsights++
 				})
 				return numInsights == tc.totalTxnInsights
-			}, 1*time.Second, 50*time.Millisecond)
+			}, 1*time.Second, 10*time.Millisecond)
 
 			// See that the insights we were expecting are the ones that
 			// arrived. We allow the provider to do whatever it needs to, so

